plugins/modules/ec2_snapshot.py

```diff
@@ -321,6 +318,8 @@ def create_snapshot(module, ec2, description=None, wait=None,
 
 
 def delete_snapshot(module, ec2, snapshot_id):
+    if module.check_mode:
+        module.exit_json(changed=True, msg='Would have deleted snapshot if not in check mode')
```

Comment:
The existing code uses the `InvalidSnapshot.NotFound` exception to tell if the snapshot existed or not.  This will break idempotency.

Since the integration tests didn't catch this it would be good to add a test that running deletion twice results in `changed` followed by  `not changed`